### PR TITLE
Let sample try to use model.test_point

### PR DESCRIPTION
**Pull request by jsalvatier**
Fixing issue #170.

### CODE REVIEW

**Comment by fonnesbeck** [comment]
This works either if you specify all initial values or none. How about retrieving the model's `test_point` and substituting any values specified in the `start` dict? This would allow for specifying some of the values, but not necessarily all.

**Comment by jsalvatier** [comment]
Sounds useful.

I'm slightly concerned that this will lead to some confusing, but I can't
think of specific cases where it would.

**Comment by fonnesbeck** [comment]
There are scenarios that I have encountered in the past where the default values are fine across a large model, except one or two (say due to a transformation of that variable), and specifying an alternative value for that variable allows the model to run.

**Comment by fonnesbeck** [comment]
See what you think of this. I just use the passed `start` value to update the `test_point`.

**Comment by jsalvatier** [comment]
Looks good to me. You might want use model.test_point.copy(), and you could
replace start = None with start = {} and get rid of a case.

**Comment by fonnesbeck** [comment]
Done.

**Comment by twiecki** [comment]
Getting some test errors like this:

``` ======================================================================
ERROR: pymc.tests.test_sampling.test_sample(10, <pymc.step_methods.metropolis.Metropolis object at 0x5ccc750>, None)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/pymc-devs/pymc/pymc/sample.py", line 49, in sample
    test_point.update(start)
TypeError: 'NoneType' object is not iterable
```

as well as

```
======================================================================
ERROR: pymc.tests.test_examples.test_examples('/home/travis/build/pymc-devs/pymc/pymc/examples/stochastic_volatility.py',)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/pymc-devs/pymc/pymc/tests/test_examples.py", line 37, in check_example
    imp.load_source('__main__', path.basename(p))
  File "stochastic_volatility.py", line 114, in <module>
    trace = sample(200, step, start, trace=model.vars + [sigma])
  File "/home/travis/build/pymc-devs/pymc/pymc/sample.py", line 44, in sample
    start = trace.point(-1)
AttributeError: 'list' object has no attribute 'point'
```

**Comment by fonnesbeck** [comment]
Runs cleanly now. I also changed one of the examples (`disaster_model`) to only partially-specify the model's starting value. Are we happy with this?

File: pymc/tests/test_sampling.py [hunk 1]
```diff
@@ -11,13 +12,24 @@
...
-        for samplr in test_samplers:
-            for n in [0, 10, 1000]:
-                yield samplr, n, step, start
+        for trace in test_traces:
+            for samplr in test_samplers:
+                for n in [0,1, 10, 300]:
+
+                    yield samplr, n, step, {}
+                    yield samplr, n, step, {}, trace
+                    yield samplr, n, step, start
```
Comment (jsalvatier) [inline pymc/tests/test_sampling.py:33]: Maybe switch this to be a partial specification of the starting point. Then I think we're good to merge.

Comment (fonnesbeck) [inline pymc/tests/test_sampling.py:33]: `simple_init` (and hence `simple_model`) yields a model with only one parameter, so it cannot be partially specified. We will have to use another model.

Comment (jsalvatier) [inline pymc/tests/test_sampling.py:33]: Oh, true. I guess I would just merge then, and we can come back to this.

**Comment by fonnesbeck** [comment]
Oops. Should probably make the same changes to `psample`.

**Pull request merged**